MyShop/src/com/examShop/UI/menu/menuLogic/MenuHandlerShopProductWork.java

```diff
@@ -0,0 +1,41 @@
+package com.examShop.UI.menu.menuLogic;
+
+import com.examShop.UI.menu.MenuCase;
+import com.examShop.UI.reader.ShopReader;
+import com.examShop.handlers.AddProductInShop;
+import com.examShop.handlers.DelelteProductInShop;
+import com.examShop.handlers.EditProductInShop;
+import com.examShop.model.shop.Shop;
+
+public class MenuHandlerShopProductWork extends MenuHandler implements AddProductInShop, DelelteProductInShop, EditProductInShop {
```

Comment:
Let me share my opinion with you. I would say that instead of implementing AddProductInShop, DelelteProductInShop, EditProductInShop, you can just have all these methods here, in the MenuHandlerShopProductWork, because you don't have any logic that is specific to any input/output there, you have logic that will probably not change in the future

So having getOptionalReader() is a great decision, because we can have a bunch of readers, but moving work with products to the 3 separate interfaces seems like an overkill to me. Interesting idea, thoug